numpy/core/arrayprint.py

```diff
@@ -394,7 +391,12 @@ def _get_format_function(data, **options):
     elif issubclass(dtypeobj, _nt.object_):
         return formatdict['object']()
     elif issubclass(dtypeobj, _nt.void):
-        return formatdict['void']()
+        # StructureFormat relies on np.void.__getitem__, so we can't use it
```

Comment:
This comment would seem helpful only if you thought `StructureFormat` should be used elsewhere (as it was originally, but which is not all that obvious). Maybe just state here something along the lines of "'void' only deals with unstructured bytes; we use StructureFormat if they are assigned to fields. Note that we cannot use StructureFormat where the void is in a union with a non-void dtype; those will be caught above and the typeset with the regular dtype (though the full dtype will be printed, to indicate to the user it is a union dtype)"

Reply:
> If you thought StructureFormat should be used elsewhere
That depends on what you think counts as a structured type:

* `issubdtype(dt, np.void) and dt.names is not None`
* `dt.names is not None`
* `issubdtype(dt, np.flexible)` (a lot of old code refers to flexible types when it means structured types)

Reply:
Agreed it is confusing, but this is one place where one can explain what it is in the code!